demo-app/build.gradle.kts

```diff
@@ -12,7 +12,6 @@ localProperties.load(FileInputStream(rootProject.file("local.properties")))
 android {
     namespace = "io.opentelemetry.android.demo"
     compileSdk = 34
-    targetSdk = 34
```

Comment:
If I understood correctly, what @bidetofevil mentioned in [this other PR](https://github.com/open-telemetry/opentelemetry-android/pull/359) is that libs shouldn't define this value, but apps should. So maybe we should keep it here?

Reply:
Ok I added it back to the `defaultConfig` block.